src/data/userDAO.ts

```diff
@@ -0,0 +1,83 @@
+import AWS, { DynamoDB } from 'aws-sdk';
+import fetch from 'node-fetch';
+
+const TABLE_NAME = 'users';
+
+export type DiscordUser = {
+    id: string;
+    username: string;
+};
+
+export type User = {
+    id: string;
+    username: string;
+    accessToken: string;
+    refreshToken: string;
+};
+
+export class UserDAO {
+    private dynamodb: DynamoDB;
+
+    constructor() {
+        this.dynamodb = new AWS.DynamoDB();
+    }
+
+    public async saveUser(accessToken: string, refreshToken: string): Promise<any> {
+        const user = await this.getDiscordUser(accessToken);
```

Comment:
I'd be remiss if I didn't point out my favorite ts feature:

`const { id, username } = await this.getDiscordUser(accessToken);`

Definitely just a style thing, but once you start destructuring it is hard to stop